src/lib.rs

```diff
@@ -94,9 +96,12 @@ impl LightningNode {
             warn!("Object storage is unhealty");
         }
 
+        // Step x. Initialize the Transaction Filter
+        let filter = Arc::new(LipaChainAccess::new(esplora_client.clone()));
+
         // Step 5. Initialize the ChainMonitor
         let chain_monitor = Arc::new(ChainMonitor::new(
-            None,
+            Some(filter.clone()),
```

Comment:
Should we follow the practice to clone `Arc` with `Arc::clone()` (see examples below)?